### PR TITLE
[CI] Use the licence sent by Artelys

### PR DESCRIPTION
**Pull request by amontoison**
It will help to test #117.

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    download(ENV["SECRET_KNITRO_LIBIOMP5"], joinpath(@__DIR__, "libiomp5.so"))
+    write_depsfile("", local_filename)
+    return
+end
+
+if get(ENV, "SECRET_KNITRO_URL", "") != ""
+    try_ci_installation()
+else
+    try_local_installation()
+end
```
Comment (odow) [inline deps/build.jl:56]: You shouldn't need this file. Just setting the environment variables should be sufficient to load KNITRO.jl

**Comment by amontoison** [comment]
I have the following error:
```julia
ERROR: LoadError: could not load library "/home/runner/.julia/packages/KNITRO/RJPC2/deps/libknitro.so"
libiomp5.so: cannot open shared object file: No such file or directory
```

**Comment by odow** [comment]
I don't think so. what happens if you add the environment variables to the test stage, not just `buildpkg`.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        run: |
+          echo "$ARTELYS_LIC_JSO" > ~/artelys_lic.txt
+      - uses: julia-actions/julia-buildpkg@v1
+        env:
+          SECRET_KNITRO_URL: ${{ secrets.SECRET_KNITRO_URL }}
+          SECRET_KNITRO_LIBIOMP5: ${{ secrets.SECRET_KNITRO_LIBIOMP5 }}
+      - uses: julia-actions/julia-runtest@v1
+        env:
+          SECRET_KNITRO_URL: ${{ secrets.SECRET_KNITRO_URL }}
+          SECRET_KNITRO_LIBIOMP5: ${{ secrets.SECRET_KNITRO_LIBIOMP5 }}
```
Comment (odow) [inline .github/workflows/ci.yml:43]: I don't know why, but I guess there is another call to `Pkg.build`, and this throws away the work we did in the previous step.

Comment (odow) [reply .github/workflows/ci.yml:43]: Great. I guess that means you can stop using your cluster for the tests

Comment (amontoison) [reply .github/workflows/ci.yml:43]: Only on Linux, I can't use the "CI" licence for Mac.

**Pull request merged**